### ISSUE TITLE
add back @add_constraints

### ISSUE THREAD

**Issue by isaacsas** [SciML/Catalyst.jl#408]
Since we now properly support sub-systems we can add it back. I think this should now just generate a `NonlinearSystem` or `ODESystem`, and then call `compose` to add it as a sub-system of the original network. If we go with `NonlinearSystem` we can special case calling `convert_system` to allow promotion to an `ODESystem` when needed.

Then we need doc updates, because in general I don't think Catalyst or ModelingToolkit can (yet!) do what people are envisioning (which seems to be system reduction via specifying conservation laws that can be used to eliminate variables). So we'll need some tutorials to show in what ways this is useful (mainly for connecting systems together).

**Comment by isaacsas** [comment]
Thinking about this more, I'm not sure that we gain much having `@add_constraints` vs. just having users use the ModelingToolkit notation to directly create a `NonlinearSystem` or `ODESystem`. I guess it saves them a bit of typing.

**Comment by ChrisRackauckas** [comment]
I think just creating the NonlinearSystem is more flexible and should be suggested.

**Comment by isaacsas** [comment]
Yeah, I’ll just add a tutorial showing some ways to use mixed systems.
